cdap-api/src/main/java/co/cask/cdap/api/schedule/AbstractCompositeTriggerInfo.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+import java.util.List;
+
+/**
+ * Base class for the composite trigger information to be passed to the triggered program.
+ */
+public abstract class AbstractCompositeTriggerInfo extends TriggerInfo {
+  List<TriggerInfo> unitTriggerInfos;
+
+  public AbstractCompositeTriggerInfo(Trigger.Type type, List<TriggerInfo> unitTriggerInfos) {
+    super(type);
+    this.unitTriggerInfos = unitTriggerInfos;
```

Comment:
cdap-api module does not include guava as dependency, so just a copy into an ArrayList?